eng/common/TestResources/SubConfig-Helpers.ps1

```diff
@@ -2,6 +2,12 @@ function BuildServiceDirectoryPrefix([string]$serviceName) {
     return $serviceName.ToUpperInvariant() + "_"
 }
 
+# If the ServiceDirectory has multiple segments use the last directory name
+# e.g. D:\foo\bar -> bar or foo/bar -> bar
+function GetServiceName([string]$serviceDirectory) {
+    return Split-Path -Leaf $serviceDirectory
```

Comment:
Is this going to be unique enough? Do we still want to trim?

Reply:
I was thinking that due to the new multi-segment service directories, however I like the brevity/simplicity/consistency of keeping the leaf. Having to reference environment variable names like `MESSAGING_AZSERVICEBUS` is going to get tedious quickly. Uniqueness is not a concern because:

1. The service name value, when not combined with other unique values, is only used for environment variable prefixes.
2. When the service name is included in the resource group name, it will either contain the local username or a random short string for CI.
    - This could be slightly confusing with groups named like `rg-internal-tf234asd45` from go, but I think the tradeoff for short environment variables is worth it. Plus we can always reverse lookup resource groups by service with the BuildId tag (or by looking at the deployed resource types).

Reply:
> The service name value, when not combined with other unique values, is only used for environment variable prefixes.

Might be worth naming the function as such so there isn't any confusion of if someone else tries use it expecting a service name.

Reply:
Yes something like that.